.travis.yml

```diff
@@ -17,26 +17,26 @@ matrix:
     - python: 2.6
       dist: trusty
       env:
-        - NOSEOPTS="" SETUPOPTS=""
+        - NOSEOPTS="--verbose" SETUPOPTS=""
     - python: 2.7
     - python: 3.4
     - python: 3.6
     - python: 3.7
       env:
         - >-
-          NOSEOPTS="--with-coverage --cover-inclusive
+          NOSEOPTS="--verbose --with-coverage --cover-inclusive
           --cover-min-percentage=60 --cover-branches
           --cover-package=azurelinuxagent --cover-xml"
           SETUPOPTS=""
 
 install:
-  - pip install -r requirements.txt
-  - pip install -r test-requirements.txt
+  - sudo pip install -r requirements.txt
+  - sudo pip install -r test-requirements.txt
 
 script:
   # future: - pylint setup.py makepkg.py azurelinuxagent/
-  - nosetests $NOSEOPTS tests
-  - if [ ! -z "$SETUPOPTS" ]; then /usr/bin/env python setup.py $SETUPOPTS; fi
+  - sudo env "PATH=$PATH" nosetests $NOSEOPTS tests
```

Comment:
does this run *all* unit tests as root?

Reply:
Yes. We currently don't have tests that need to be run as a normal user in order to succeed.